Esri/arcgis-maps-sdk-swift-samples
# [New] Set reference scale

## Description

This PR implements `Set reference scale` in `Maps` category.
URL to README: [README](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-SetReferenceScale/Shared/Samples/Set%20reference%20scale)

## Linked Issue(s)

- `swift/issues/5403`

## How To Test

- Tap "Map Settings".
- Select a value in the "Reference Scale" picker to change the reference scale of the map.
- Tap "Set to Reference Scale" to update the map's scale to the selected reference scale.
- Tap "Layer" and toggle whether a layer should honor the reference scale.

## Screenshots

![set-reference-scale](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/fc0253f9-ab5f-41a3-bd60-b0fc52720d52)
